lib/src/utils/resolveRichTextToNodes.ts

```diff
@@ -109,7 +109,15 @@ export const resolveNode = (
     const { text, marks } = node;
 
     if (marks) {
-      let marked: ComponentNode[] = [{ content: text }];
+      let marked: ComponentNode[] = stringResolver
+        ? [
+            {
+              content: text,
+              ...stringResolver(text),
+            },
+          ]
+        : [{ content: text }];
```

Comment:
both branch sides has same base of `[{ content: text }]` while the `stringResolver` usage is conditional to the left side, so maybe this could be simplified to the:

```js
let marked: ComponentNode[] = [
  {
    content: text,
    ...stringResolver?.(text),
  },
];
```

did not try the code, but in case it does not work it might need to additionally add conditional to not spread undefined: `...(stringResolver?.('d') || {}`

Reply:
correct, simplified the implementation

Additionally, handled a corner case when both schema text node resolver and `textResolver` are provided, to not override each other. See the test cases